### PR TITLE
fixed skew transformations

### LINKED ISSUE #11884: Transform 'skewY' affect to dimensions
### Render

```jsx
<View style={{flex: 1, padding: 70}}>
       <View style={{flex: 0.5, backgroundColor: 'red', transform:  [{skewY: '0deg'}]}}/>
       <View style={{flex: 0.5, backgroundColor: 'blue', transform: [{skewY: '50deg'}]}}/>
</View>
```

### Result

Result

### Expected result (with fixing via scaleX: 1.555)
CorrectResult

### PR DESCRIPTION
**Pull request by MaxGraey**
Motivation:
fix #11884 issue

@sahrens @vjeux

### CODE REVIEW

**Comment by vjeux** [comment]
@jordwalke you wrote that code, could you review it please? Thanks!

**Comment by MaxGraey** [comment]
Hi, @jordwalke. I will try in short to explain what was wrong.
Consider the previous transformation matrix for skewY.
#### Was
| cos(a) | sin(a) |    0   |   0   |
|:------:|:------:|:------:|:-----:|
|    0   |    1   |    0   |   0   |
|    0   |    0   |   1    |    0  |
|   tx   |   ty   |   tz   |   1   ||

Yes, this visually produce skewing transform but it slightly incorrect. This way affect horizontal scale as well. See [this](https://github.com/facebook/react-native/issues/11884)

### Now with PR

|    1   | tan(a) |    0   |   0   |
|:------:|:------:|:------:|:-----:|
|    0   |    1   |    0   |   0   |
|    0   |    0   |   1    |   0   |
|   tx   |   ty   |   tz   |   1   ||

According to [www.w3.org/css-transforms](https://www.w3.org/TR/css-transforms-1/#SkewDefined)

Only one differance React Native used column major matrix style, so we change ```m[0][1]``` instead ```m[1][0]```.

**Comment by vjeux** [comment]
Thanks for the thorough explanation, that sounds good to me. I just clicked the land button, so unless something goes wrong, it's going to ship in the next tens of minutes! Thanks!

File: Libraries/ReactNative/YellowBox.js [hunk 1]
```diff
@@ -69,10 +69,13 @@ if (__DEV__) {
     warn.apply(console, arguments);
     updateWarningMap.apply(null, arguments);
   };
+<<<<<<< HEAD
```
Comment (vjeux) [inline Libraries/ReactNative/YellowBox.js]: I think something went wrong with your rebase

Comment (MaxGraey) [inline Libraries/ReactNative/YellowBox.js]: @vjeux  My last fork was 5 days ago, so I try to remote update to actual base, but anyway I still have unresolved 2 conflicts. What would you recommend?

**Comment by vjeux** [comment]
Since your diff is so small ( https://github.com/facebook/react-native/pull/11886/commits/7aefe360533617d226705d9ad56f8a2e27b3ca50 ), I would recommend removing this whole branch, applying it back manually and sending a force update. The git commands should look something like:

```js
git checkout origin/master;
git branch -D master;
git checkout -b master;
// apply the changes
git commit
// copy paste the description from github
git push -f MaxGraey
```

In the future, I recommend that you create a branch with

```js
git checkout -b fix_skew
```

this way you don't need to work on master and you can have multiple of them going on.

**Comment by vjeux** [comment]
Looks like there are still a lot of spurious changes :(

**Comment by MaxGraey** [comment]
@vjeux I choose second approach

[New PR from fix-skew](https://github.com/facebook/react-native/pull/11992)

**Pull request closed without merging**